app/services/configuration_repository_service.rb

```diff
@@ -9,25 +9,36 @@ class ConfigurationRepositoryService
     include FastlaneCI::Logging
     include FastlaneCI::GitHubHandler
 
+    # An octokit client authenticated with the onboarding user's API token
+    #
+    # @return [Octokit::Client]
+    attr_reader :onboarding_user_client
+
+    # An octokit client authenticated with the bot user's API token
+    #
     # @return [Octokit::Client]
-    attr_reader :client
+    attr_reader :bot_user_client
 
     # Instantiates new `ConfigurationRepositoryService` class
     #
     # @param  [ProviderCredential] provider_credential
     def initialize(provider_credential: nil)
-      @client = Octokit::Client.new(access_token: provider_credential.api_token)
+      @onboarding_user_client = Octokit::Client.new(access_token: provider_credential.api_token)
+      @bot_user_client = Octokit::Client.new(access_token: FastlaneCI.env.ci_user_api_token)
     end
 
-    # Creates a remote repository if it does not already exist, complete with
-    # the expected remote files `user.json` and `projects.json`
-    def create_private_remote_configuration_repo
-      github_action(client) do
-        # TODO: Handle the common case of when provided account can't create a new private repo
-        client.create_repository(repo_name, private: true) unless configuration_repository_exists?
-        create_remote_json_file("users.json", json_string: serialized_users)
-        create_remote_json_file("projects.json")
-      end
+    # Sets up the `fastlane.ci` configuration repository, with the necessary
+    # configuration files
+    #
+    #   i.   Creates the remote repository
+    #   ii.  Adds the bot user as a collaborator and accepts the invitation as
+    #        the bot user
+    #   iii. Creates the necessary remote configuration files as the bot user
+    #
+    def setup_private_configuration_repo
+      create_private_remote_configuration_repo
+      add_bot_user_as_collaborator
+      create_remote_configuration_files
```

Comment:
Would love to see some tests for this behavior.

Reply:
You're welcome to write them. Stubbing out octokit and verifying remote GitHub files are written from an external collaborater (just added) to a private remote GitHub repository (just created) is a non-trivial task.

Reply:
I'll need to stub out a fair bit of octokit and the GitHub API (requests, responses). I might not get to it until tomorrow, but I'll try to write a lot of reusable helpers so that stubbing the GitHub API for other tests is easy too.

Reply:
Oh and also, don't worry about tests in this PR, I think we should just make sure it works in a clean environment, merge, and then add those tests I mentioned above.